framework/db/BaseActiveRecord.php

```diff
@@ -1200,7 +1200,7 @@ public static function populateRecord($record, $row)
      * For example, by creating a record based on the value of a column,
      * you may implement the so-called single-table inheritance mapping.
      * @param array $row row data to be populated into the record.
-     * @return static the newly created active record
+     * @return self the newly created active record
```

Comment:
The type is correct now.

Reply:
But if I override the return type with my own, then Psalm gives this error
```
The inherited return type 'yii\db\BaseActiveRecord&static' for yii\db\BaseActiveRecord::instantiate is more specific than the implemented return type for Product::instantiate 'Smartphone|Tablet|Laptop'
```

Reply:
In case of single table inheritance, I don't think so. That's exceptional situation.